uplink/config.go

```diff
@@ -150,3 +156,47 @@ func (c Config) GetEncryptionScheme() storj.EncryptionScheme {
 		BlockSize: int32(c.Enc.BlockSize),
 	}
 }
+
+// LoadEncryptionKey loads the encryption key stored in the file pointed by
+// filepath.
+//
+// An error is file is not found or there is an I/O error.
+func LoadEncryptionKey(filepath string) (key *storj.Key, error error) {
+	if filepath == "" {
+		return &storj.Key{}, nil
+	}
+
+	file, err := os.Open(filepath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("not found key file %q", filepath)
+		}
+
+		return nil, err
+	}
+
+	defer func() { err = errs.Combine(err, file.Close()) }()
+
+	// TODO: if/v3-1541#3 should we check if the file contains a values of length
+	// storj.KeySize??
+	key = &storj.Key{}
+	if _, err := file.Read(key[:]); err != nil && err != io.EOF {
+		return nil, err
+	}
+
+	return key, nil
+}
+
+// UseOrLoadEncryptionKey return an encryption key from humanReadableKey when
+// it isn't empty otherwise try to load the key from the file pointed by
+// filepath calling LoadEncryptionKey function.
+func UseOrLoadEncryptionKey(humanReadableKey string, filepath string) (*storj.Key, error) {
+	if humanReadableKey != "" {
+		var key storj.Key
+		copy(key[:], humanReadableKey)
```

Comment:
Shouldn't we use `storj.NewKey(humanReadableKey)` here?

Reply:
Not for this one, this is not a passphrase, what's wanted here is to use an encryption key passed by parameter not generating one from the value passed.

Reply:
I think we've discussed in one of the PRs you have closed that we don't want to use the human readable key directly as the root encryption key, but we want to derive the encryption key from it using PBKDF2 or something else. This is what `storj.NewKey` does. So we should use it here.

Reply:
The problem is that I haven't been able to find a way for generating file to store the encryption key and being able to clean it up on the integration tests.

I asked about how I could do it, but the response was kind of "there is not way so far" and that's one of the reasons for not having I/O on `uplink` package. Slack: https://storj.slack.com/archives/CAQV0AE2Z/p1557937208391100

This is also mentioned in the PR comment.

On the other hand, the goal of the issue wasn't about that, it was to have the encryption key outside of the `config.yaml` file, so it already contains more functionality on what the issue describe. I'm not saying that we should do more changes, but I think that new issues must be created for tackling them or at least, recreate the issue to indicate what it has to be done, because on each PR that I've sent new things are appearing and making impossible to estimate the issue on every sprint that it gets dragged.

Reply:
OK. So let's drop the PBKDF2 key derivation from this PR. Let `storj.NewKey` just do `copy(key[:], humanReadableKey)` like before, but let's have this logic just in `storj.NewKey` and nowhere else. This means that whenever we want to get `storj.Key` from the human-readable key, we should call `storj.NewKey`. The encryption file should store the human-readable key in plain text (no ending newline).

We will figure out in a separate PR how to properly do the key derivation from the human-readable key.